### PR TITLE
add two new routers and change get_res router

### PR DESCRIPTION
**Pull request by TwinIsland**
1. **get_base**: return the base address of the project
2. **upload_file**: upload file in specific catagory 

change:
1. **get_res**: now need to point out which resource type (rtype) to get

### CODE REVIEW

File: engine/doc/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<head>
```
Comment (TwinIsland) [inline engine/doc/index.html:1]: the index page for engine server

File: engine/module/project_manager.py [hunk 2]
```diff
@@ -46,8 +46,8 @@ class ProjectManager:
     """
 
     def __init__(self):
-        self.__config_res = None
-        self.__base = None
+        self.__config_res: dict = {}
```
Comment (TwinIsland) [inline engine/module/project_manager.py:49]: initialized but not set optional

File: engine/module/project_manager.py [hunk 3]
```diff
@@ -57,15 +57,18 @@ def init_project(self, base_dir: str, config_dir: str):
...
 
         self.__base = base_dir
         config = ConfigLoader(config_dir)
         self.__config_res = config.resources()
+        project_base_dir = config.project()["projects_base"]
+        self.__base = os.path.join(project_base_dir, self.__base)
 
-        for i in config.resources().keys():
-            res_path_abs = os.path.join(base_dir, config.resources()[i])
+        if not file_utils.check_folder_valid(self.__base):
```
Comment (TwinIsland) [inline engine/module/project_manager.py:67]: hardcoding due to resources config change

File: engine/server.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    @router_exception_handler
+    def get_base_dir(self) -> ReturnDict:
+        to_return = {"base": self.__project_manager.get_base_dir()}
+        return ReturnDict(status=STATUS.OK, content=to_return)
+
+
+router_utils = RouterUtils()
+
+
+@app.get("/", include_in_schema=False)
```
Comment (TwinIsland) [inline engine/server.py:159]: index page for engine

File: engine/service.ini [hunk 5]
```diff
@@ -11,6 +11,13 @@ default_suffix=log
 background_dir=resources/background
 music_dir=resources/music
 character_dir=resources/character
+max_size=52428800
```
Comment (TwinIsland) [inline engine/service.ini:14]: upload file options

**Review by TwinIsland — COMMENTED** [review]
see comment above

**Review by Tesseract22 — APPROVED** [review]
All good.

**Pull request merged**